logger.go

```diff
@@ -0,0 +1,10 @@
+package process
+
+type Logger interface {
+	WithFields(LogFields) Logger
+	Info(string, ...interface{})
+	Warning(string, ...interface{})
+	Error(string, ...interface{})
+}
```

Comment:
Weren't we talking about having a `BaseLogger` with just a `Log` method to make it easier to implement custom loggers?

Reply:
The actual logger in defined in nacelle, this is just to have an interface that nacelle uses. The user-facing logger is much more basic.